src/__tests__/roadwork.ts

```diff
@@ -0,0 +1,46 @@
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import mongoose from 'mongoose';
+
+import { createRoadwork, findRoadwork } from '../models/Roadwork';
+
+const mongod = new MongoMemoryServer();
+
+const databaseName = 'test_roadwork';
+
+const name = 'name1';
+const description = 'description1';
+
+describe('testing roadwork', () => {
+  beforeAll(async () => {
+    const uri = await mongod.getUri(databaseName);
+
+    await mongoose.connect(uri, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+      keepAlive: true,
+    });
+  });
+
+  it('creates a roadwork database object', async () => {
+    const result = await createRoadwork(null, {
+      name,
+      description,
+    });
+
+    expect(result.name).toEqual(name);
+    expect(result.description).toEqual(description);
+  });
+
+  it('finds the previously created roadwork', async () => {
+    const result = await findRoadwork();
+    expect(result).toHaveLength(1);
+    expect(result[0]?.name).toEqual(name);
+    expect(result[0]?.description).toEqual(description);
+  });
+
+  afterAll(async () => {
+    await mongoose.connection.dropDatabase();
+    await mongoose.connection.close();
+    mongod.stop();
+  });
+});
```

Comment:
why have you decided to put the __tests_ folder inside of src?
I think it can have advantages to separate __tests__ from src with having them next to each other when working with the folder structure of the project for example.

Reply:
@donni106  and i discussed it, and decided to try out the concept of having the test files adjacent to the tested files in the same folder, with the same name + `.test.ts` as an ending.